staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -659,16 +654,22 @@ func validateCustomResourceDefinitionValidation(customResourceValidation *apiext
 			allowDefaults:            opts.allowDefaults,
 			requireValidPropertyType: opts.requireValidPropertyType,
 		}
+
 		allErrs = append(allErrs, ValidateCustomResourceDefinitionOpenAPISchema(schema, fldPath.Child("openAPIV3Schema"), openAPIV3Schema, true)...)
 
-		if opts.requireStructuralSchema {
+		if opts.requireStructuralSchema && len(allErrs) == 0 {
```

Comment:
why was this moved here? this change means that *any* other validation error in the CRD will mask all structural schema validation errors, then suddenly reveal them all once other errors are resolved. that's not a great user experience.

Reply:
I have reverted this. Now we always try to convert to structural, and as before ignore the error if allErrs is non-empty. If `err==nil`, we continue with the other checks (that depend on structural schemas).